rust/arrow/src/ipc/convert.rs

```diff
@@ -859,7 +859,7 @@ mod tests {
             0, 0,
         ];
         let ipc2 = ipc::root_as_message(&bytes[..]).unwrap();
-        let schema2 = ipc.header_as_schema().unwrap();
+        let schema2 = ipc2.header_as_schema().unwrap();
```

Comment:
reading the code, it does appear that the intent of this test is to compare the schema from two different IPC message (aka `schema2` should be coming from the `ipc2` message) as @gangliao  has done here. 


The test seems to have failed however https://github.com/apache/arrow/pull/9764/checks?check_run_id=2161212421


```
thread 'ipc::convert::tests::schema_from_bytes' panicked at 'assertion failed: `(left == right)`
  left: `Schema { endianness: Little, fields: Some([Field { name: Some("field1"), nullable: false, type_type: Int, type_: Int { bitWidth: 32, is_signed: false }, dictionary: None, children: Some([]), custom_metadata: None }]), custom_metadata: None, features: None }`,
 right: `Schema { endianness: Little, fields: Some([Field { name: Some("field1"), nullable: false, type_type: Int, type_: Int { bitWidth: 32, is_signed: false }, dictionary: None, children: None, custom_metadata: None }]), custom_metadata: Some([]), features: None }`', arrow/src/ipc/convert.rs:864:9
```

The difference is that `schema` has `children: Some([])` and `schema2` has `children: None`

Reply:
yes, I didn't find a good way to fix it.

Reply:
maybe we can assert the individual fields that are the same. Something like

```
assert_eq!(schema.endianess, schema2.endianess)
assert_eq!(schema.fields(), schema2.fields())
...
```